From 204a4c94c5c56c9fd6998942f46e55b978e709cd Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stefan=20Br=C3=BCns?= <stefan.bruens@rwth-aachen.de>
Date: Sat, 14 Sep 2019 22:13:13 +0200
Subject: [PATCH] Fix link errors when compiling with LTO enabled

When libsigrok is compiled with link-time-optimization, the linker
stumbles over the section named the same as the function. Whether this
is a compiler/linker bug or a coding error is unknown.

Renaming the special section (to the same as already used on OS X) avoids
the problem.

This fixes bug #1416.
---
 src/libsigrok-internal.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libsigrok-internal.h b/src/libsigrok-internal.h
index a0f00b24..c57f2058 100644
--- a/src/libsigrok-internal.h
+++ b/src/libsigrok-internal.h
@@ -279,7 +279,7 @@ struct zip_stat;
 #ifdef __APPLE__
 #define SR_DRIVER_LIST_SECTION "__DATA,__sr_driver_list"
 #else
-#define SR_DRIVER_LIST_SECTION "sr_driver_list"
+#define SR_DRIVER_LIST_SECTION "__sr_driver_list"
 #endif
 
 /**
-- 
2.23.0

